scripts/manage_pes

```diff
@@ -148,456 +89,355 @@ USAGE, OPTIONS, AND EXAMPLES
+            compsetfile = p1
+            pesxmlfile = p2
+        except Exception, e:
+            pass
+
+        # Read in given xml file
+        if not os.path.isfile(self.filename):
+            logger.critical(" File %s not found", self.filename)
+            sys.exit(1)
+
+        newxml = Pes(infile=self.filename)
+
+        gridname, machname, pesizename, compsetname = self.checkFile(newxml)
+
+        # Is there already an entry with matching grid, machine, compset and
+        # pesize?
+        # If so, query if the new settings should replace the current ones
+        # If not - just add the new settings
+        pesxml = Pes(infile=pesxmlfile)
+        sys.stdout.write("------------------------------------------- \n")
+        sys.stdout.write(" Pes set by     : %s \n"
+                         % pesxmlfile)
+        sys.stdout.write(" Compsets set by: %s \n" % compsetfile)
+        sys.stdout.write("------------------------------------------- \n\n")
+        matches = self.getMatches(pesxml, grid=gridname, mach=machname,
+                                  compset=compsetname)
+        newmatch = self.getMatches(newxml, grid=gridname,
+                                   mach=machname, compset=compsetname)[0]
+
+        if len(matches) > 0:
+            match = matches[0]
+
+            sys.stdout.write(" The following pe-layout match already exists\n")
+            self.printMatches([match])
+            sys.stdout.write(" The new values would be\n")
+            self.printMatches([newmatch])
+            override = raw_input(" Do you want to override the match with"
+                                 " your pe-layout [yes/no] (default is no)\n")
+            if override.lower() != "y" and override.lower() != "yes":
+                return
+
+            # remove old node
+            pesxml.root.remove(match)
+
+        # add new node
+        pesxml.add_child(newmatch)
+
+
+        newfilename = "%s-%s" % (pesxml.filename,
+                        datetime.datetime.now().strftime("%d%b%Y-%H%M%S"))
+
+        sys.stdout.write("\n now writing the new pes list to %s\n" %
+                         newfilename)
+        sys.stdout.write("Please carefully review and/or diff the new file "
+                         "against the\n")
+        sys.stdout.write("original, and if you are satisfied with the "
+                         "changes, move \n")
+        sys.stdout.write("%s to \n" % newfilename)
+        sys.stdout.write("as in with the unix command: \n")
+        sys.stdout.write("mv %s %s\n\n" % (newfilename, pesxml.filename))
+
+        pesxml.write(newfilename)
+
+
+    def queryPes(self):
+        """
+        Read the xml files, query the object, and print the user's choice
+        of output.
+        """
+
+        logger.debug("starting queryPes")
+        filesxml = Files()
+        if self.setby is None:
+            comps = ["allactive", "cam", "cice", "cism", "clm", "drv", "pop"]
```

Comment:
The specific component names (cam, cice, etc ) should be acquired from files.get_components("COMPSETS_SPEC_FILE") .